openedx/core/djangoapps/programs/utils.py

```diff
@@ -90,6 +91,9 @@ def __init__(self, site, user, enrollments=None, uuid=None):
             # We can't use dict.keys() for this because the course run ids need to be ordered
             self.course_run_ids.append(enrollment_id)
 
+        self.entitlements = list(CourseEntitlement.entitlements_for_user(self.user))
```

Comment:
Just a reminder that the expired_at date on an entitlement does not mean that entitlement is not still active with a potential enrollment.  Is this case solved by the fact that there would be an enrolled course_run?

I would check with @jlajoie  on the edge cases.

Reply:
If the user has an existing enrollment in a course in the program, then the program will have that as a link to the user, and therefore we don't also need to check the user's entitlements for engagement in the program.